### PR TITLE
[FR] Rename RecognizeIdDocuments for RecognizeIdentityDocuments

### LINKED ISSUE #20686: Consider renaming IdDocuments to IdentificationDocuments

### PR DESCRIPTION
**Pull request by maririos**
After feedback from architects and in agreement with v-team and service team, we decided to rename `RecognizeIdDocuments` for `RecognizeIdentityDocuments`.
This change affects: src, tests, samples, and documentation.

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/20686

### CODE REVIEW

**Review by catalinaperalta — APPROVED** [review]
LGTM!

**Review by kinelski — APPROVED** [review]
LGTM

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md [hunk 1]
```diff
@@ -622,7 +622,7 @@ This project has adopted the [Microsoft Open Source Code of Conduct][code_of_con
 [recognize_receipts]: https://github.com/Azure/azure-sdk-for-net/tree/master/sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample3_RecognizeReceipts.md
 [recognize_business_cards]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample9_RecognizeBusinessCards.md
 [recognize_invoices]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample10_RecognizeInvoices.md
-[recognize_id_documents]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample11_RecognizeIdDocuments.md
+[recognize_identity_documents]: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample11_RecognizeIdentityDocuments.md
```
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:625]: yeah, no idea

**Pull request merged**